File: zaptorch/src/main/java/com/pyamsoft/zaptorch/app/settings/AccessibilityRequestDialog.java
/*
 * Copyright 2016 Peter Kenji Yamanaka
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.pyamsoft.zaptorch.app.settings;

import android.app.Dialog;
import android.content.Intent;
import android.os.Bundle;
import android.provider.Settings;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;
import android.support.v7.app.AlertDialog;
import com.pyamsoft.zaptorch.R;
import com.pyamsoft.zaptorch.ZapTorch;

public class AccessibilityRequestDialog extends DialogFragment {

  @SuppressWarnings("WeakerAccess") @NonNull final Intent accessibilityServiceIntent =
      new Intent(Settings.ACTION_ACCESSIBILITY_SETTINGS);

  @NonNull @Override public Dialog onCreateDialog(Bundle savedInstanceState) {
    return new AlertDialog.Builder(getActivity()).setTitle("Enable ZapTorch AccessibilityService")
        .setMessage(R.string.explain_accessibility_service)
        .setPositiveButton("Let's Go", (dialogInterface, i) -> {
          getActivity().startActivity(accessibilityServiceIntent);
          dismiss();
        })
        .setNegativeButton("No Thanks", (dialogInterface, i) -> dismiss())
        .create();
  }

  @Override public void onDestroy() {
    super.onDestroy();
    ZapTorch.getRefWatcher(this).watch(this);
  }
}

File: zaptorch/src/main/java/com/pyamsoft/zaptorch/settings/AccessibilityRequestDialog.kt
/*
 * Copyright (C) 2018 Peter Kenji Yamanaka
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.pyamsoft.zaptorch.settings

import android.app.Dialog
import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import androidx.appcompat.app.AlertDialog
import com.pyamsoft.pydroid.ui.app.fragment.ToolbarDialog
import com.pyamsoft.zaptorch.R

class AccessibilityRequestDialog : ToolbarDialog() {

  private val accessibilityServiceIntent = Intent(Settings.ACTION_ACCESSIBILITY_SETTINGS)

  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
    return AlertDialog.Builder(requireActivity())
        .setTitle("Enable ZapTorch AccessibilityService")
        .setMessage(R.string.explain_accessibility_service)
        .setPositiveButton("Let's Go") { _, _ ->
          requireActivity().startActivity(accessibilityServiceIntent)
          dismiss()
        }
        .setNegativeButton("No Thanks") { _, _ -> dismiss() }
        .create()
  }
}
